src/pages/materials/materials.js

```diff
@@ -1,29 +1,79 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
-import Typography from '@material-ui/core/Typography';
+import parse from 'html-react-parser';
 
-import { useStyles } from './materials.style';
+import { useStyles } from './materials.style.js';
+import { getBusinessPageByCode } from '../../redux/business-pages/business-pages.actions';
+import AwesomeSlider from 'react-awesome-slider';
+import withAutoplay from 'react-awesome-slider/dist/autoplay';
+import 'react-awesome-slider/dist/styles.css';
+
+import { carouselMaterialInterval, IMG_URL } from '../../configs';
+import { getPatterns } from '../../../src/redux/pattern/pattern.actions';
+import clsx from 'clsx';
+import { getImage } from '../../utils/imageLoad';
+
+const AutoplaySlider = withAutoplay(AwesomeSlider);
 
 const Materials = () => {
+  const [setImage] = useState([]);
   const dispatch = useDispatch();
-  const styles = useStyles();
-  const { language } = useSelector(({ Language }) => ({
-    language: Language.language
-  }));
+  const { materialsPage, language, patterns, images } = useSelector(
+    ({ BusinessPages, Language, Pattern, HomePageSlider }) => ({
+      materialsPage: BusinessPages.pages.materials,
+      language: Language.language,
+      patterns: Pattern.list,
+      loading: HomePageSlider.loading,
+      images: HomePageSlider.images
+    })
+  );
 
   useEffect(() => {
     window.scrollTo(0, 0);
+    dispatch(getBusinessPageByCode('materials'));
+    dispatch(getPatterns());
   }, [dispatch]);
 
-  const titleText = language
-    ? 'This page has not been developed yet'
-    : 'Ця сторінка ще не розроблена';
+  useMemo(() => {
+    patterns.images &&
+      patterns.images.forEach((item) => {
+        getImage(patterns.images.medium)
+          .then((src) => setImage((prev) => [...prev, src]))
+          .catch((badSrc) => setImage((prev) => [...prev, badSrc]));
+      });
+  }, [images]);
 
+  const materialPageText =
+    materialsPage.text && parse(materialsPage.text[language].value);
+  const styles = useStyles();
+  const imagesForSlider = patterns.map((pattern) => (
+    <div
+      className={'sliderImage'}
+      style={{ width: '100%', height: '100%' }}
```

Comment:
You can set styles for class name in `materials.styles.js`.

Reply:
done